src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -1396,7 +1396,15 @@ private CSharpSyntaxNode GetBindingRoot(CSharpSyntaxNode node)
                 {
                     case SyntaxKind.ThisConstructorInitializer:
                     case SyntaxKind.BaseConstructorInitializer:
+                        return current;
                     case SyntaxKind.ArrowExpressionClause:
+                        // If this is an arrow expression on a local function statement, then our bindable root is actually our parent syntax as it's
+                        // a statement in a function. If this is returned directly in IOperation, we'll end up with a separate tree.
+                        if (current.ParentOrStructuredTriviaParent != null && current.ParentOrStructuredTriviaParent.Kind() == SyntaxKind.LocalFunctionStatement)
+                        {
+                            return current.ParentOrStructuredTriviaParent;
```

Comment:
>ParentOrStructuredTriviaParent [](start = 43, length = 30)

Parent?

Reply:
Or simply continue the loop?

---
In reply to: [177942626](https://github.com/dotnet/roslyn/pull/25782#discussion_r177942626) [](ancestors = 177942626)

Reply:
It seemed clearer to me to just return here, instead of inserting a continue. If we match a syntax here, we will return.

---
In reply to: [178117247](https://github.com/dotnet/roslyn/pull/25782#discussion_r178117247) [](ancestors = 178117247,177942626)

Reply:
> It seemed clearer to me to just return here, instead of inserting a continue. If we match a syntax here, we will return.

The code is structured so that we decide whether the node we are looking at is a binding root. If not, we continue, we don't try to make decision whether the parent is a good candidate instead.  

---
In reply to: [178128501](https://github.com/dotnet/roslyn/pull/25782#discussion_r178128501) [](ancestors = 178128501,178117247,177942626)